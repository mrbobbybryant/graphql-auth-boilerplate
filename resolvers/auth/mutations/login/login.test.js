import { login } from './index';

const get = jest.fn();
const User = {
  get,
};

const compare = jest.fn();
const jwt = jest.fn();
const bcrypt = {
  compare,
};

const testLogin = login(User, bcrypt, jwt);

describe('login Tests', () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });

  test('Should return an error if user is not found.', async () => {
    get.mockReturnValueOnce(null);
    try {
      const result = await testLogin(
        {},
        {
          input: {
            email: 'bobby@crossfield.com',
            password: '1234',
          },
        },
        {},
      );
      throw 'This should not run';
    } catch (error) {
      const info = JSON.parse(error.message);
      expect(info.message).toEqual('Invalid Email.');
      expect(info.code).toEqual(400);
      expect(info.field).toEqual('email');
    }
  });

  test('Should return an error if the password is not valid.', async () => {
    get.mockReturnValueOnce({
      id: 1,
      email: 'bobby@crossfield.com',
      password: '1234',
    });
    compare.mockReturnValueOnce(false);
    try {
      const result = await testLogin(
        {},
        {
          input: {
            email: 'bobby@crossfield.com',
            password: '1234',
          },
        },
        {},
      );
      throw 'This should not run';
    } catch (error) {
      const info = JSON.parse(error.message);
      expect(info.message).toEqual('Invalid Password.');
      expect(info.code).toEqual(400);
      expect(info.field).toEqual('password');
    }
  });

  test('Should work.', async () => {
    get.mockReturnValueOnce({
      id: 1,
      email: 'bobby@crossfield.com',
      password: '1234',
    });
    compare.mockReturnValueOnce(true);
    jwt.mockReturnValueOnce('IOIJIO');

    const result = await testLogin(
      {},
      {
        input: {
          email: 'bobby@crossfield.com',
          password: '1234',
        },
      },
      {},
    );

    expect(result).toMatchSnapshot();
  });
});
