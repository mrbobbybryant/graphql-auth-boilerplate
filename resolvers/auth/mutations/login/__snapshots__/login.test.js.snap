// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login Tests Should work. 1`] = `
Object {
  "email": "bobby@crossfield.com",
  "id": 1,
  "password": "1234",
  "token": "IOIJIO",
}
`;
